dnsutil/zone.go

```diff
@@ -0,0 +1,209 @@
+package dnsutil
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"sort"
+	"strings"
+
+	"github.com/miekg/dns"
+)
+
+type zoneGenData struct {
+	Origin     string
+	DefaultTtl uint32
+	Records    []dns.RR
+}
+
+func (z *zoneGenData) Len() int      { return len(z.Records) }
+func (z *zoneGenData) Swap(i, j int) { z.Records[i], z.Records[j] = z.Records[j], z.Records[i] }
+func (z *zoneGenData) Less(i, j int) bool {
+	a, b := z.Records[i], z.Records[j]
```

Comment:
I prefer this sort order because it groups hostnames in a way that makes it easier for people to see patterns: Grouping all records in a subdomain next to each other.  I see the value of having it sorted in DNSSEC order, but it isn't as readable, but the code is structured so it would be easy to abstract out the ordering code and create a WriteZoneFileOrdered() function that lets you supply a custom sort function.